src/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Windows.cs

```diff
@@ -156,14 +159,11 @@ public static SocketError Send(SafeCloseSocket handle, IList<ArraySegment<byte>>
             }
             finally
             {
-                if (objectsToPin != null)
+                for (int i = 0; i < objectsToPin.Length; ++i)
```

Comment:
If you make the aforementioned change to use a const value in the stackalloc or to use ArrayPool, you'd want to change this to `count` rather than `objectsToPin.Length`.

Reply:
done